UnityProject/Assets/Scripts/Items/Others/Knife.cs

```diff
@@ -27,40 +27,40 @@ public bool WillInteract(InventoryApply interaction, NetworkSide side)
 
 		return true;
 	}
+
 	public void ServerPerformInteraction(InventoryApply interaction)
 	{
 		//is the target item cuttable?
 		ItemAttributesV2 attr = interaction.TargetObject.GetComponent<ItemAttributesV2>();
 		Ingredient ingredient = new Ingredient(attr.ArticleName);
 		GameObject cut = CraftingManager.Cuts.FindRecipe(new List<Ingredient> { ingredient });
-		if (cut && interaction.TargetObject.GetComponent<Stackable>() != null && ingredient.requiredAmount
-			== interaction.TargetObject.GetComponent<Stackable>().Amount)
-		{
-			Inventory.ServerDespawn(interaction.TargetSlot);
 
-			SpawnResult spwn = Spawn.ServerPrefab(CraftingManager.Cuts.FindOutputMeal(cut.name),
-			SpawnDestination.At(), 1);
-
-			if (spwn.Successful)
-			{
-				Inventory.ServerAdd(spwn.GameObject, interaction.TargetSlot);
-			}
+		if (cut == null)
+		{
+			Chat.AddExamineMsgFromServer(interaction.Performer, "You can't cut this.");
+			return;
 		}
-		else if (cut)
+
+		if (interaction.TargetObject.TryGetComponent(out Stackable stackable) &&
+				stackable.Amount != ingredient.requiredAmount)
 		{
-			Inventory.ServerDespawn(interaction.TargetSlot);
+			Chat.AddExamineMsgFromServer(interaction.Performer, "Not enough or too much of the ingredient.");
+			return;
+		}
 
-			SpawnResult spwn = Spawn.ServerPrefab(CraftingManager.Cuts.FindOutputMeal(cut.name),
-			SpawnDestination.At(), 1);
+		Cut(interaction, cut);
+	}
 
-			if (spwn.Successful)
-			{
-				Inventory.ServerAdd(spwn.GameObject, interaction.TargetSlot);
-			}
-		}
-		else
+	private void Cut(InventoryApply interaction, GameObject cut)
```

Comment:
Codacy found an issue: [Rename the parameter 'cut' so that it does not duplicate the method name.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6335489)